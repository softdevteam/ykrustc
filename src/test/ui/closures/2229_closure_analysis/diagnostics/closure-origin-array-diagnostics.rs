#![feature(capture_disjoint_fields)]
//~^ WARNING: the feature `capture_disjoint_fields` is incomplete
//~| `#[warn(incomplete_features)]` on by default
//~| see issue #53488 <https://github.com/rust-lang/rust/issues/53488>

// Test that array access is not stored as part of closure kind origin

fn expect_fn<F: Fn()>(_f: F) {}

fn main() {
    let s = [format!("s"), format!("s")];
    let c = || { //~ ERROR expected a closure that implements the `Fn`
        let [_, _s] = s;
    };
    expect_fn(c);
}
