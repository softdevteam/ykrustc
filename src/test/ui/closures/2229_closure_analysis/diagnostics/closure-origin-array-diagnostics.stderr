warning: the feature `capture_disjoint_fields` is incomplete and may not be safe to use and/or cause compiler crashes
  --> $DIR/closure-origin-array-diagnostics.rs:1:12
   |
LL | #![feature(capture_disjoint_fields)]
   |            ^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: `#[warn(incomplete_features)]` on by default
   = note: see issue #53488 <https://github.com/rust-lang/rust/issues/53488> for more information

error[E0525]: expected a closure that implements the `Fn` trait, but this closure only implements `FnOnce`
  --> $DIR/closure-origin-array-diagnostics.rs:12:13
   |
LL |     let c = || {
   |             ^^ this closure implements `FnOnce`, not `Fn`
LL |         let [_, _s] = s;
   |                       - closure is `FnOnce` because it moves the variable `s` out of its environment
LL |     };
LL |     expect_fn(c);
   |     --------- the requirement to implement `Fn` derives from here

error: aborting due to previous error; 1 warning emitted

For more information about this error, try `rustc --explain E0525`.
